R/derive_param_clinbenefit.R

```diff
@@ -0,0 +1,273 @@
+#' Derive a Clinical Benefit Parameter
+#'
+#' Add a clinical benefit/disease control parameter to the input dataset.
+#'
+#' @details
+#' Clinical benefit/disease control is first identified for looking for subjects
+#' having response status, and then derived for subjects that have at least one
+#' evaluable non-PD response assessment prior to first PD (Progressive Disease)
+#' (i.e., responses exclusive of `NA`, `NE`, `ND`, and `PD`) and after a specified
+#' amount of time from a reference date (`ref_start_window`).
+#'
+#' \enumerate{
+#'   \item The input dataset (`dataset`) is restricted to the observations matching
+#'   `filter_source` and to observations before or at the date specified by `source_pd`.
+#'
+#'   \item This dataset is further restricted to exclude response assessments of `NA`,
+#'   `NE`, `ND`, and `PD`, missing response assessments, and those less than
+#'   `ref_start_window` after `reference_date`. The earliest assessment by `ADT`
+#'   is then selected.
+#'
+#'   \item The dataset identified by `dataset` in `source_resp` is restricted
+#'   according to its `filter` argument. The variable corresponding to the `date`
+#'   parameter of `source_resp` is considered together with `ADT` from the
+#'   previous step.
+#'
+#'   \item For the observations being added to `dataset`, `ADT` is set to the earlier
+#'   of the first assessment date representing an evaluable non-PD assessment prior
+#'   to first PD, or the date representing the start of response.
+#'
+#'   \item For the observations being added to `dataset`, `AVALC` is set to:
+#'
+#'     - `Y` for those subjects in the `dataset` meeting the criteria for clinical
+#'     benefit above
+#'
+#'     - `N` for subjects not meeting the clinical benefit criteria in `dataset`
+#'     or the dataset identified in `source_resp`
+#'
+#'     - `N` for subjects present in `dataset_adsl` but not present in `dataset`
+#'     or the dataset identified in `source_resp`.
+#'
+#'   \item `AVAL` is derived using `AVALC` as input to the function specified in
+#'   `aval_fun`.
+#'
+#'   \item The variables specified by `set_values_to` are added to the new observations
+#'   with values equal to the values specified in the same.
+#'
+#'   \item The new observations are added to `dataset`.
+#'  }
```

Comment:
It should be mentioned which variables are kept from the source observations.

Reply:
updated